src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/FrameComponent.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    /// <summary>
+    /// The class for showing UI module
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public abstract class FrameComponent : BaseComponent
+    {
+        /// <summary>
+        /// Gets the display status of a component.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Thrown when component type is already added to the component.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public DisplayStatus DisplayStatus
+        {
+            get
+            {
+                Interop.CBApplication.NativeDisplayStatus status;
+                Interop.CBApplication.ErrorCode err = Interop.CBApplication.BaseFrameGetDisplayStatus(Handle, out status);
+                if (err != Interop.CBApplication.ErrorCode.None)
+                    throw new InvalidOperationException("Fail to get display status : err(" + err + ")");
+
+                return (DisplayStatus)status;
+            }
+        }
+
+        /// <summary>
+        /// Gets the frame component's window information.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public IWindowInfo WindowInfo { get; internal set; }
```

Comment:
How change like below, current code is violation of encapsulation

``` c#
        public IWindowInfo WindowInfo { get; private set; }
        public void OnCreate()
        {
            WindowInfo = CreateWindowInfo();
        }
        public abstract IWindowInfo CreateWindowInfo();

     // caller
     fc.OnCreate();
     if (fc.WindowInfo == null)
                return IntPtr.Zero;
     Interop.CBApplication.BaseFrameCreateWindow(out winHandle, fc.WindowInfo.ResourceId, IntPtr.Zero);
```

Reply:
Hmm... this way, we cannot explicitly force to make a window in OnCreate callback.
Maybe we need specific documentation.

But, as you say current code violate encapsulation so, I will discuss it with scrum product owner @pjh9216
He is on vacation until next Monday.

Reply:
> Hmm... this way, we cannot explicitly force to make a window in OnCreate callback.
Why do you think so?
`public void OnCreate()` is not virtual, so can't modify on subclass
And CreateWindowInfo method was newly added and it inforce creation of WindowInfo
`public abstract IWindowInfo CreateWindowInfo();`

Reply:
If you want to provide the way to add some code when created, 
You can provide a event like `Created`

Reply:
CreateWindowInfo is added and OnCreate does not return IWindow now.
Also, WindowInfo property is removed. The WindowInfo property is unnecessary.